test/test-device-ping1d.cpp

```diff
@@ -0,0 +1,113 @@
+#include <ping-device-ping1d.h>
+#include <ping-message-common.h>
+#include <ping-message-ping1d.h>
+#include <ping-port-linux.h>
+
+#include <stdio.h>
+
+#define EXENAME "test-device-ping1d"
+
+int main(int argc, char* argv[]) {
+    printf(" ~* " EXENAME " *~\n");
+
+    if (argc < 2 || argc > 2) {
+        printf("usage: " EXENAME " <path to port>\n");
+        printf("ex: " EXENAME " /dev/ttyUSB0\n");
+        return 1;
+    }
+
+    const char* portFileName = argv[1];
+
+    PingPortLinux port = PingPortLinux(portFileName);
+    Ping1d device      = Ping1d(port);
+
+    printf("initializing\n");
+
+    if (!device.initialize(100)) {
+        printf("device initialization failed\n");
+    } else {
+        printf("pass\n");
+        printf("Device Type %d id %d hardware revision %d\n", device.device_type, device.device_id, device.device_revision);
+        printf("Ping Protocol v%d.%d.%d\n", device.version_major, device.version_minor, device.version_patch);
+        printf("Device Firmware v%d.%d.%d\n", device.firmware_version_major, device.firmware_version_minor, device.firmware_version_patch);
+    }
+
+    printf("set device id\n");
+    if (!device.set_device_id(1)) {
+        printf("failed to set device id\n");
+    } else {
+        printf("pass\n");
+    }
```

Comment:
std::cout << "set device id" << (device.set_device_id(1) ? success_str : "failed to set device id");